File: client-go/ColorContainer_test.go
package animatedledstrip

import "testing"

func TestColorContainer_AddColor(t *testing.T) {
	c := ColorContainer{}
	c.AddColor(0xFF)
	if c.colors[0] != 0xFF {
		t.Fail()
	}
	c.AddColor(0xFFFF)
	if c.colors[0] != 0xFF || c.colors[1] != 0xFFFF {
		t.Fail()
	}
}

func TestColorContainer_Json(t *testing.T) {
	c := ColorContainer{}
	c.AddColor(0xFF00FF)
	c.AddColor(0xFFFF)
	if c.Json() != `{"colors":[16711935,65535]}` {
		t.Fail()
	}
}

File: client-dart/test/color_container_test.dart
import 'package:animatedledstripclient/animatedledstripclient.dart';
import 'package:test/test.dart';

void main() {
  test("Default Construction", () {
    var cc = ColorContainer();
    expect(cc.colors, isEmpty);
  });

  test("Add Color", () {
    var cc = ColorContainer();
    expect(cc.colors, isEmpty);
    cc.addColor(0xFF);
    cc.addColor(0xFF00);
    expect(cc.colors, isNotEmpty);
    expect(cc.colors[0], 0xFF);
    expect(cc.colors[1], 0xFF00);
  });

  test("JSON", () {
    var cc = ColorContainer()
        ..addColor(0xFF0000)
        ..addColor(0xFF00);

    expect('{"colors":[16711680, 65280]}', cc.json());
    
    var cc2 = ColorContainer();
    expect('{"colors":[]}', cc2.json());
  });
}